### PR TITLE
Synchronise the game start between players.

### PR DESCRIPTION
**Pull request by Scrabouillmax**
- There is a lobby of players.
- One player pushes start: a START_GAME event goes to the back
- The back broadcasts the actual game start event to all clients (with the countdown time)
- After the countdown, the back sends the first song.

### CODE REVIEW

File: backend/src/controllers/game_controller.ts [hunk 1]
```diff
@@ -66,9 +66,8 @@ export class GameController extends Controller {
     }
 
     // save new game
-    const now: number = new Date().getTime();
     const game: Game = new Game();
-    game.startDate = req.body.startDate !== undefined ? toDate(req.body.startDate) || new Date(now + 30 * 1000) : new Date(now + 30 * 1000);
```
Comment (Scrabouillmax) [inline backend/src/controllers/game_controller.ts]: I do that as we don't use this anymore to determine when the game starts (the player pushing the button is the signal)

File: backend/src/controllers/game_controller.spec.ts [hunk 2]
```diff
@@ -125,7 +125,7 @@ describe("Test POST, PUT, GET", () => {
       .send();
     expect(res.status).toEqual(200);
     expect(res.body.id).toBe(game.id);
-    expect(res.body.startDate).toBe("2020-01-21T15:55:25.436Z");
+    expect(typeof res.body.startDate).toEqual("string");
```
Comment (2016floreza) [inline backend/src/controllers/game_controller.spec.ts]: I do not believe this is a sufficiant test to know if it is a date. The old version was better even if it wasn't optimal.

File: frontend/src/pages/ConnectedPages/Game/Countdown/index.tsx [hunk 3]
```diff
@@ -18,7 +24,7 @@ const Countdown: React.FC<ICountdown> = ({ setStep }: ICountdown) => {
         setStep(IGameEnum.QUIZZ);
       }
     }, 1000);
-  }, [countdown]);
+  }, [countdown, setStep]);
```
Comment (2016floreza) [inline frontend/src/pages/ConnectedPages/Game/Countdown/index.tsx]: I know there is a warning but I do not like seen the setStep here

File: frontend/src/pages/ConnectedPages/Game/index.tsx [hunk 4]
```diff
@@ -24,6 +25,8 @@ const Game: React.FC<RouteComponentProps> = () => {
     setStep(IGameEnum.QUIZZ);
   };
 
+  const countdownMs = querystring.parse(location.search).countdownMs as string;
```
Comment (2016floreza) [inline frontend/src/pages/ConnectedPages/Game/index.tsx]: This is great ! I need to use this when checking urls before.

File: backend/src/controllers/game_controller.spec.ts (hunk 2 above)
Comment (DavidRobertAnsart) [reply backend/src/controllers/game_controller.spec.ts]: On pourrais faire 
```
expect(/^\d{4}-[0-1]\d-[0-3]\dT(\d{2}:?){3}.\d{3}Z$/.test(res.body.startDate)).toBeTruthy();
```

File: frontend/src/pages/ConnectedPages/Game/Countdown/index.tsx (hunk 3 above)
Comment (Scrabouillmax) [reply frontend/src/pages/ConnectedPages/Game/Countdown/index.tsx]: I don't understand ^^

**Pull request merged**